awx/ui_next/src/screens/Organization/OrganizationList/OrganizationList.jsx

```diff
@@ -114,118 +50,141 @@ class OrganizationsList extends Component {
         {
           data: { actions },
         },
-      ] = await promises;
-      this.setState({
-        actions,
-        itemCount: count,
-        organizations: results,
-        selected: [],
-      });
-    } catch (err) {
-      this.setState({ contentError: err });
+      ] = await Promise.all([
+        OrganizationsAPI.read(params),
+        loadOrganizationActions(),
+      ]);
+      setItemCount(count);
+      setOrganizations(results);
+      setOrgActions(actions);
+      setSelected([]);
+    } catch (error) {
+      setContentError(error);
     } finally {
-      this.setState({ hasContentLoading: false });
+      setHasContentLoading(false);
     }
-  }
+  };
 
-  render() {
-    const {
-      actions,
-      itemCount,
-      contentError,
-      hasContentLoading,
-      deletionError,
-      selected,
-      organizations,
-    } = this.state;
-    const { match, i18n } = this.props;
+  const loadOrganizationActions = () => {
+    if (orgActions) {
+      return Promise.resolve({ data: { actions: orgActions } });
+    }
+    return OrganizationsAPI.readOptions();
+  };
 
-    const canAdd =
-      actions && Object.prototype.hasOwnProperty.call(actions, 'POST');
-    const isAllSelected =
-      selected.length === organizations.length && selected.length > 0;
+  const handleOrgDelete = async () => {
+    setHasContentLoading(true);
+    try {
+      await Promise.all(selected.map(({ id }) => OrganizationsAPI.destroy(id)));
+    } catch (error) {
+      setDeletionError(error);
+    } finally {
+      await loadOrganizations(location);
+    }
```

Comment:
Should this reset the selected list to an empty array? (more complicated if an error occurs... would we only de-select ones that successfully delete?)

Reply:
Yep - we set the selected list back to an empty array inside `loadOrganizations`. If _that_ request fails, the page will reload into a contentError so we don't need to worry about selection state in that case.

I agree with you that more precise handling of the individual failure states could get complex. At this point in time, partial failures on multi-deletes will still result in an empty selection list via the reload. I view this as reasonable-enough behavior for now.